src/components/onboarding-modal/index.js

```diff
@@ -19,24 +19,26 @@ class OnboardingModal extends Component {
     this.closeModal = this.closeModal.bind(this)
   }
 
-  async componentWillMount() {
-    const { fetchSteps, onboarding, wallet } = this.props
+  async componentDidMount() {
+    const { fetchSteps } = this.props
     await fetchSteps()
 
-    if (!onboarding.stepsCompleted && !wallet.address) {
+    window.setTimeout(() => {
       this.userProgress()
-    }
+    }, 500)
```

Comment:
Half a second wait is probably required for the Dapp to register if wallet is present and `wallet` property is available through redux. On some machines it might take less than 500ms, on the other hand on others(that are really slow) it might take more. 

Another approach would be to mark when a wallet is initialised ( https://github.com/OriginProtocol/origin-dapp/blob/master/src/reducers/Wallet.js ): 
```
const initialState = {
  address: undefined,
  ethBalance: '0',
  ognBalance: '0',
  initialised: false
}

export default function Wallet(state = initialState, action = {}) {
  switch (action.type) {
  case WalletConstants.INIT_SUCCESS:
    return { ...state, address: action.wallet, initialised: true}
...
```

and call `userProgress` from `componentWillUpdate` only once `wallet.initilised === true`